app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/EmailController.scala

```diff
@@ -79,16 +69,9 @@ class EmailController @Inject() (
         if (groupEnrolments.exists(_.service == service.enrolmentKey))
           // user has specified service
           if (service.code.equalsIgnoreCase(appConfig.standaloneServiceCode))
-            existingEoriForUserOrGroup(user, groupEnrolments) match {
-              case Some(eori) =>
-                sessionCache.saveEori(Eori(eori.id)).map(
-                  _ => Redirect(EnrolmentAlreadyExistsController.enrolmentAlreadyExistsForGroupStandalone(service))
-                )
-              case None =>
-                Future.successful(
-                  Redirect(EnrolmentAlreadyExistsController.enrolmentAlreadyExistsForGroupStandalone(service))
-                )
-            }
+            existingEoriForUserOrGroup(user, groupEnrolments)
+              .traverse(eori => sessionCache.saveEori(Eori(eori.id)))
+              .map(_ => Redirect(EnrolmentAlreadyExistsController.enrolmentAlreadyExistsForGroupStandalone(service)))
```

Comment:
Are the two different paths unit tested?

It looks good but I did have to have a refresher on traverse haha